pkg/ee/validation/machine/providers.go

```diff
@@ -42,17 +42,27 @@ import (
 	"k8c.io/kubermatic/v2/pkg/resources"
 	"k8c.io/kubermatic/v2/pkg/resources/certificates"
 
+	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/resource"
+	k8stypes "k8s.io/apimachinery/pkg/types"
 	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-func GetMachineResourceUsage(ctx context.Context, userClient ctrlruntimeclient.Client, machine *clusterv1alpha1.Machine,
-	caBundle *certificates.CABundle) (*ResourceDetails, error) {
+const ClusterLabel = "system/cluster"
+
+func GetMachineResourceUsage(ctx context.Context,
+	seedClient ctrlruntimeclient.Client,
+	userClient ctrlruntimeclient.Client,
+	machine *clusterv1alpha1.Machine,
+	caBundle *certificates.CABundle,
+) (*ResourceDetails, error) {
 	config, err := types.GetConfig(machine.Spec.ProviderSpec)
 	if err != nil {
 		return nil, fmt.Errorf("failed to read machine.spec.providerSpec: %w", err)
 	}
 
+	clusterID := machine.Spec.Labels[ClusterLabel]
```

Comment:
Is it safe to assume that the label exists without checking? I suggest checking and returning error in case label does not exist.